### PR TITLE
Migrate Java Project to Gradle + Switch to Oracle GraalVM for JDK 17 With Build Process (And Small Improvements)

### PR DESCRIPTION
**Pull request by Minionguyjpro**
This PR will migrate the project from Apache Maven to Gradle, with quicker building speeds. The action will also use Oracle GraalVM instead now for building (which is quicker than Eclipse Temurin/Adoptium). It also includes the generated shaded/uber JAR with a successful build in the ``AP-TEST-SERVER`` directory as an artifact. If you go to the Actions tab and click the first ``Build`` action, you'll find an ``Artifacts`` section at the bottom called ``AutoPlug-Client`` which is the artifact. If you download it, you'll get a ZIP file with the JAR inside. You can then test it, and if it works, you can upload it to GitHub and you're good to go (don't forget to update the version number first in the ``build.gradle`` file  and use that artifact if you wish to make a new release!). Apart from that I made small improvements, you can check them at any time!

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,10 +15,11 @@ jobs:
...
+      - name: Set up Oracle GraalVM for JDK 17
+        uses: graalvm/setup-graalvm@v1
         with:
-          java-version: 1.17
-      - name: Build with Maven
-        run: mvn -B package --file pom.xml
+          java-version: '17'
+          distribution: 'graalvm'
+      - name: Build with Gradle
+        run: ./gradlew build shadowJar -x test --no-daemon
```
Comment (Osiris-Team) [inline .github/workflows/test.yml:25]: Excludes tests, however should do the exact opposite and run only tests.

File: .gitignore [hunk 2]
```diff
@@ -1,29 +1,21 @@
-**/target/
-**/AP-TEST-SERVER/
-/autoplug
-/test
+.gradle
```
Comment (Osiris-Team) [inline .gitignore:1]: Keep the old gitignore stuff

File: README.md [hunk 3]
```diff
@@ -71,11 +71,11 @@ Details:
 - Written in [Java](https://java.com/),
```
Comment (Osiris-Team) [inline README.md:71]: Keep run configs XML files above too, change the command to use gradle instead.
Also do the same for the other run configs by intellij below.
(Somehow couldn't add a file comment thus added it here)

File: README.md [hunk 4]
```diff
@@ -71,11 +71,11 @@ Details:
...
   with [JDK 8](https://www.oracle.com/java/technologies/javase/javase-jdk8-downloads.html), inside
   of [IntelliJ IDEA](https://www.jetbrains.com/idea/)
-- Built with [Maven](https://maven.apache.org/), profiles: `clean package` to create the `./AP-TEST-SERVER/AutoPlug-Client.jar` 
+- Built with [Gradle](https://gradle.org/), profiles: `./gradlew build shadowJar` (Linux and UNIX systems) or `.\gradlew.bat build shadowJar` (Windows systems) from project root directory to create the `./AP-TEST-SERVER/AutoPlug-Client.jar`
```
Comment (Osiris-Team) [inline README.md:74]: Rename profiles to command

File: build.gradle [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    archiveFileName = 'AutoPlug-Client.jar'
+    // Output into directory for manual tests
+    destinationDirectory = file("AP-TEST-SERVER")
+    dependsOn generateAutoplugProperties
+}
+
+// Configure jar task
+jar {
+    from sourceSets.main.output
+    archiveFileName = 'original-AutoPlug-Client.jar'
```
Comment (Osiris-Team) [inline build.gradle:49]: I see `AutoPlug-Client.jar` is used a few times, maybe create it as string/property?

File: build.gradle [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+
+// Responsible for running all tests before building.
+test {
+    useJUnitPlatform()
+}
+
+tasks.withType(JavaCompile) {
+    options.encoding = 'UTF-8'
+}
+
```
Comment (Osiris-Team) [inline build.gradle:138]: This file looks pretty good! Remind me to check the versions of the dependencies once I'm at my PC.

**Review by Osiris-Team — CHANGES_REQUESTED** [review]
This looks pretty good thanks! A few small things tho.

File: build.gradle (hunk 5 above)
Comment (Minionguyjpro) [reply build.gradle]: Okay @Osiris-Team, I changed all of them to point to ``rootProject.name`` which is defined as ``AutoPlug-Client`` in the ``settings.gradle`` file.

File: README.md (hunk 3 above)
Comment (Minionguyjpro) [reply README.md]: @Osiris-Team, do you have a guide for that? And you mean the .run directory, right? That XML forwards to a lot of Maven-specific blocks.

Comment (Minionguyjpro) [reply README.md]: Okay thanks. I'll try it out tomorrow in IntelIJ IDEA CE on my Xubuntu environment and see whether I can get the run tasks back but for Gradle.

**Comment by Minionguyjpro** [comment]
Do I still need to add the .run stuff @Osiris-Team? Everything is ready except this yet, am trying to configure it now, though at some points I am not completely sure with all options. Like with that Maven configuration, it has one configuration to skip tests but what else?

**Comment by Minionguyjpro** [comment]
> Do I still need to add the .run stuff @Osiris-Team? Everything is ready except this yet, am trying to configure it now, though at some points I am not completely sure with all options. Like with that Maven configuration, it has one configuration to skip tests but what else?

Nevermind, got it. It's actually pretty easy. Am finishing the last configurations right now.

**Comment by Minionguyjpro** [comment]
@Osiris-Team Is there a way that I can manually try running my configuration XML files or run them?

**Comment by Minionguyjpro** [comment]
@Osiris-Team Done! Should be ready to get merged into the upstream (a.k.a. this repository).

**Comment by Osiris-Team** [comment]
@Minionguyjpro will do once another recent issue related to backup uploads is fixed. Seems to have something to do with dependencies.

**Comment by Minionguyjpro** [comment]
> @Minionguyjpro will do once another recent issue related to backup uploads is fixed. Seems to have something to do with dependencies.
> 
> Thanks!

Tell me when you get that fixed. If it's indeed a dependency, tell me what you changed so I can do the same for the build.gradle. Plus, sync changes if it's in something else than the dependencies in the pom.xml.

**Review by Osiris-Team — APPROVED** [review]

**Pull request merged**